[PyTorch] return `decltype(auto)` from getItem

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117569
* #117568

This allows getItem to take advantage of the nicer (sometimes-const-reference) return type from `List::get() const` added in the previous diff.

Differential Revision: [D52809097](https://our.internmc.facebook.com/intern/diff/D52809097/)

swolchok committed:
[PyTorch] return `decltype(auto)` from getItem

This allows getItem to take advantage of the nicer (sometimes-const-reference) return type from `List::get() const` added in the previous diff.

Differential Revision: [D52809097](https://our.internmc.facebook.com/intern/diff/D52809097/)

[ghstack-poisoned]

iseeyuan reviewed (approved):
Thanks for updating this return type in a nicer way!

swolchok committed:
Update on "[PyTorch] return `decltype(auto)` from getItem"


This allows getItem to take advantage of the nicer (sometimes-const-reference) return type from `List::get() const` added in the previous diff.

Differential Revision: [D52809097](https://our.internmc.facebook.com/intern/diff/D52809097/)

[ghstack-poisoned]

swolchok commented:
@pytorchbot merge
